py/dml/c_backend.py

```diff
@@ -1597,12 +1597,10 @@ def generate_init_static_vars(device):
     start_function_definition(
         'void _init_static_vars(%s *_dev)' % (crep.structtype(device),))
     out('{\n', postindent = 1)
-    with crep.DeviceInstanceContext():
-        for v in device.staticvars:
-            if v.init:
-                assert isinstance(v.init, Initializer)
-                var = mkStaticVariable(v.site, v)
-                v.init.assign_to(var, var.sym.type)
+    for (_, init) in dml.globals.static_vars:
+        if init:
+            for line in init.splitlines(keepends=True):
+                out(line)
```

Comment:
I think `out` handles multiline strings correctly, no need to split

Reply:
Nope, it doesn't. It fails at indenting them properly.

Reply:
Check out `Output.out`, it's actually rather obvious when looking at the code.

... I suppose your next comment is "fix `Output.out` instead then". Ergh, I guess I could _try_, but I don't know how much of a pain that would be or what kind of ramifications it would have.

Reply:
OK, fair enough, it seems to support printing part of a line but not multiple lines.
Multiline `out` is uncommon so no need to fix.